packages/core/versioning/core-version-http-server/src/version_http_toolkit.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  IRouter,
+  RouteConfig,
+  RouteMethod,
+  RequestHandler,
+  RouteValidatorFullConfig,
+  RequestHandlerContextBase,
+} from '@kbn/core-http-server';
+
+type RqCtx = RequestHandlerContextBase;
+
+/** A set of type literals to determine accepted versions */
+export type Version = '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '10';
```

Comment:
```suggestion
export type Version = `${number}`;
```

Reply:
Happy to change to this, if we do go with monotonically increasing numbers we should add a runtime check that version > 0.